File: Ollama/Introduction/tuto_ollama.mjs
// Source: 
import ollama from 'ollama'

// Fonctions de base
console.log(await ollama.list())
console.log(await ollama.show({model: 'qwen2.5:3b'}))
console.log(await ollama.pull({model: 'nomic-embed-text'}))

// Prompt simple
let res1 = await ollama.generate({
  model: 'qwen2.5:3b', 
  prompt: 'Quel langage de programmation fonctionnel choisir?'
  })

console.log(res1.response)

// Chat
let response = await ollama.chat({
    model: 'qwen2.5:3b',
    messages: [{ role: 'user', content: 'Qui es-tu?' }],
  })

console.log(response.message.content)

File: Ollama/Introduction/tuto_ollama.py
# Source: 
import ollama

# Fonctions de base
ollama.list()
ollama.show('qwen2.5:3b')
ollama.pull('nomic-embed-text')

# Prompt simple
res1 = ollama.generate(
    model='qwen2.5:3b', 
    prompt='Quel langage de programmation fonctionnel choisir?'
    )

print(res1.response)

# Chat: prompt plus complexe
res2 = ollama.chat(
    model='qwen2.5:3b', 
    messages=[{'role': 'user', 'content': 'Qui es-tu?'}]
    )

print(res2.message.content)

# Embedding
texte = ["C'est un texte.", "Une phrase courte."]

embedding = ollama.embed(
    model = "bge-m3",
    input= texte
)

print(embedding.embeddings)